treeherder/model/derived/jobs.py

```diff
@@ -1189,45 +1110,106 @@ def _load_ref_and_job_data_structs(
                 tier = tier_info["tier"]
 
-        job_placeholders.append([
-            job_guid,
-            signature_hash,
-            None,                   # idx:2, job_coalesced_to_guid,
-            revision,               # idx:3, replace with result_set_id
-            build_platform.id,
-            machine_platform.id,
-            machine.id,
-            option_collection_hash,
-            job_type.id,
-            product.id,
-            who,
-            reason,
-            job.get('result', 'unknown'),  # idx:12, this is typically an int
-            state,
-            self.get_number(job.get('submit_timestamp')),
-            self.get_number(job.get('start_timestamp')),
-            self.get_number(job.get('end_timestamp')),
-            0,                      # idx:17, replace with average job duration
-            tier,
-            job_guid,
-            get_guid_root(job_guid)  # will be the same except for ``retry`` jobs
-        ])
-
-        artifacts = job.get('artifacts', [])
-
-        has_text_log_summary = False
+        try:
+            duration = JobDuration.objects.values_list(
+                'average_duration', flat=True).get(
+                    repository__name=self.project, signature=signature_hash)
+        except JobDuration.DoesNotExist:
+            duration = 0
+
+        # insert the job unconditionally (we will just ignore the operation
+        # if it does not exist)
+        self.execute(
+            proc='jobs.inserts.create_job_data',
+            debug_show=self.DEBUG,
+            placeholders=[
+                [
+                    job_guid,
+                    signature_hash,
+                    None,                   # idx:2, job_coalesced_to_guid,
+                    result_set_id,
+                    build_platform.id,
+                    machine_platform.id,
+                    machine.id,
+                    option_collection_hash,
+                    job_type.id,
+                    product.id,
+                    who,
+                    reason,
+                    result,
+                    state,
+                    self.get_number(job_datum.get('submit_timestamp')),
+                    self.get_number(job_datum.get('start_timestamp')),
+                    self.get_number(job_datum.get('end_timestamp')),
+                    duration,
+                    tier,
+                    job_guid,
+                    get_guid_root(job_guid)  # will be the same except for ``retry`` jobs
+                ]
+            ],
+            executemany=True)
+
+        # by default we should try to update the "root" object
+        guid_root = get_guid_root(job_guid)
+        ds_job_ids = self.get_job_ids_by_guid([guid_root])
+        if ds_job_ids:
+            ds_job_id = ds_job_ids[guid_root]['id']
+        else:
+            ds_job_id = self.get_job_ids_by_guid([job_guid])[job_guid]['id']
+
+        if state != 'pending':
```

Comment:
Doesn't this mean that if the first time we see a job is when it's, say, "running" then we'll insert it AND update it?